op-deployer/pkg/deployer/state/intent.go

```diff
@@ -32,35 +33,313 @@ func (d DeploymentStrategy) Check() error {
+	if err != nil {
+		return nil, fmt.Errorf("error getting superchain config: %w", err)
+	}
+
+	proxyAdmin, _ := standard.ManagerOwnerAddrFor(l1ChainId)
+	guardian, _ := standard.GuardianAddressFor(l1ChainId)
+
+	superchainRoles := &SuperchainRoles{
+		ProxyAdminOwner:       proxyAdmin,
+		ProtocolVersionsOwner: common.Address(*superCfg.Config.ProtocolVersionsAddr),
+		Guardian:              guardian,
+	}
+
+	return superchainRoles, nil
+}
+
+func (c *Intent) SetInitValues(l2ChainIds []common.Hash) error {
+	switch c.IntentConfigType {
+	case IntentConfigTypeCustom:
+		return c.setCustomValues(l2ChainIds)
+
+	case IntentConfigTypeStandard:
+		return c.setStandardValues(l2ChainIds)
+
+	case IntentConfigTypeStrict:
+		return c.setStrictValues(l2ChainIds)
+
+	case IntentConfigTypeStrictOverrides:
+		return c.setStrictValues(l2ChainIds)
+
+	case IntentConfigTypeStandardOverrides:
+		return c.setStandardValues(l2ChainIds)
+
+	case IntentConfigTypeTest:
+		return c.setTestValues(l2ChainIds)
+
+	default:
+		return fmt.Errorf("intent config type not supported")
+	}
+}
+
+// Sets all Intent fields to their zero value with the expectation that the
+// user will populate the values before running 'apply'
+func (c *Intent) setCustomValues(l2ChainIds []common.Hash) error {
+	c.L1ContractsLocator = &artifacts.Locator{Tag: "undefined"}
+	c.L2ContractsLocator = &artifacts.Locator{Tag: "undefined"}
+
+	c.SuperchainRoles = &SuperchainRoles{}
+	for _, l2ChainID := range l2ChainIds {
+		c.Chains = append(c.Chains, &ChainIntent{
+			ID: l2ChainID,
+		})
+	}
+	return nil
+}
+
+func (c *Intent) setStandardValues(l2ChainIds []common.Hash) error {
+	superchainRoles, err := getStandardSuperchainRoles(c.L1ChainID)
+	if err != nil {
+		return fmt.Errorf("error getting standard superchain roles: %w", err)
+	}
+	c.SuperchainRoles = superchainRoles
+
+	c.L1ContractsLocator = artifacts.DefaultL1ContractsLocator
+	c.L2ContractsLocator = artifacts.DefaultL2ContractsLocator
+
+	for _, l2ChainID := range l2ChainIds {
+		c.Chains = append(c.Chains, &ChainIntent{
+			ID:                       l2ChainID,
+			Eip1559DenominatorCanyon: standard.Eip1559DenominatorCanyon,
+			Eip1559Denominator:       standard.Eip1559Denominator,
+			Eip1559Elasticity:        standard.Eip1559Elasticity,
+		})
+	}
+	return nil
+}
+
+// Same as setStandardValues, but also sets l2 Challenger and L1ProxyAdminOwner
+// addresses to standard values
+func (c *Intent) setStrictValues(l2ChainIds []common.Hash) error {
+	if err := c.setStandardValues(l2ChainIds); err != nil {
+		return err
+	}
+
+	challenger, _ := standard.ChallengerAddressFor(c.L1ChainID)
+	l1ProxyAdminOwner, _ := standard.ManagerOwnerAddrFor(c.L1ChainID)
+	for chainIndex := range c.Chains {
+		c.Chains[chainIndex].Roles.Challenger = challenger
+		c.Chains[chainIndex].Roles.L1ProxyAdminOwner = l1ProxyAdminOwner
+	}
+	return nil
+}
+
+func (c *Intent) setTestValues(l2ChainIds []common.Hash) error {
```

Comment:
I think it'd be cleaner to leave this in the test package rather than exposing it here.